gph/python/ripser_interface.py

```diff
@@ -251,6 +252,21 @@ def _ideal_thresh(dm, thresh):
     return min([enclosing_radius, thresh])
 
 
+def _is_prime_and_larger_than_2(x, N):
+    """Test whether 2 < x <= N is prime. Returns False when x is 2."""
+    if not x % 2 or x > N:
+        return False
+
+    # https://stackoverflow.com/questions/2068372/fastest-way-to-list-all-primes-below-n-in-python/3035188#3035188
+    sieve = [True] * N
+    for i in range(3, min(x, int(math.sqrt(N))) + 1, 2):
+        if sieve[i]:
+            sieve[i * i::2 * i] = \
+                [False] * int(((N - i * i - 1) / (2 * i) + 1))
```

Comment:
```suggestion
    sieve = [True] * (x + 1)
    for i in range(3, int(math.sqrt(x)) + 1, 2):
        if sieve[i]:
            sieve[i * i::2 * i] = \
                [False] * ((x - i * i) // (2 * i) + 1)
```

This version actually works when `x=N`, and makes the code faster/cleaner.